devito/types/petsc.py

```diff
@@ -210,3 +218,37 @@ def solver_parameters(self):
         return self._solver_parameters
 
     func = Reconstructable._rebuild
+
+
+class Mock(sympy.Function, Reconstructable):
+
+    """
+    Represents a 'mock' dependency to ensure distinct iteration loops.
+
+    For example, the mat-vec action iteration loop is to be isolated from the
```

Comment:
How would this behave if you had multiple solves? Can you group the mat-vec actions in one loop and the expression loop in the other?

Reply:
No because each PETScSolve involves a unique matrix-vector action, which, along with its corresponding iteration loop, is placed within a distinct callback function so they can't be grouped together.